### PR TITLE
Update ConfidenceInterval class constructor parameters

### LINKED ISSUE #936: Defining `self.min_rel_change` in confidence.py as variable
Might be consider `self.min_rel_change` as variable with default value instead of hard coded

https://github.com/lmfit/lmfit-py/blob/493107b85dac867d19a3a74e9d26946aab386a7c/lmfit/confidence.py#L199

### PR DESCRIPTION
**Pull request by Anselmoo**
Defining `self.min_rel_change` in confidence.py as variable Fixes #936

#### Description

This pull request updates the constructor parameters of the ConfidenceInterval class in `confidence.py`. Specifically, it adds a new parameter called `min_rel_change` with a default value of **1e-5**. This change addresses issue #936.

###### Type of Changes

- [ ] Bug fix
- [x] New feature
- [ ] Refactoring / maintenance
- [ ] Documentation / examples

###### Verification 
Have you

- [ ] included docstrings that follow PEP 257?

- [ ] referenced existing Issue and/or provided relevant link to mailing list?

- [x] verified that existing tests pass locally?

- [ ] verified that the documentation builds locally?

- [ ] squashed/minimized your commits and written descriptive commit messages?

- [ ] added or updated existing tests to cover the changes?
- [ ] updated the documentation and/or added an entry to the release notes (doc/whatsnew.rst)?
- [ ] added an example?

### CODE REVIEW

**Comment by Tillsten** [comment]
I fully agree with addition. I think a test showing that changing the parameter does something useful would be nice.

File: lmfit/confidence.py [hunk 1]
```diff
@@ -163,7 +163,8 @@ class ConfidenceInterval:
     """Class used to calculate the confidence interval."""
 
     def __init__(self, minimizer, result, p_names=None, prob_func=None,
-                 sigmas=None, trace=False, verbose=False, maxiter=50):
+                sigmas=None, trace=False, verbose=False, min_rel_change=1e-5,
+                maxiter=50):
```
[the remark this answers is not in the document]
Comment (reneeotten) [reply lmfit/confidence.py:167]: also please write a more meaningful commit message

[the remark this answers is not in the document]
Comment (Anselmoo) [reply lmfit/confidence.py:167]: I agree and please also check 

```suggestion
def conf_interval(minimizer, result, p_names=None, sigmas=None, trace=False,
                  maxiter=200, verbose=False, prob_func=None, 
                  min_rel_change=1e-5):
```

it also has to be updated

[the remark this answers is not in the document]
Comment (Anselmoo) [reply lmfit/confidence.py:167]: One more thing, I had the issue with `flake8`, and i had lint it, so a few functions inputs looks different. @reneeotten is that still ok? Othwerise, I will revert it

[the remark this answers is not in the document]
Comment (reneeotten) [reply lmfit/confidence.py]: no, that's not okay - please revert. The way it was passed the "pre-commit" checks so there is no reason to change them. We want minimal diffs so no unneeded/unrelated changes please.

[the remark this answers is not in the document]
Comment (Anselmoo) [reply lmfit/confidence.py]: Sorry, I struggled with `flake8` indentations.

File: lmfit/confidence.py [hunk 2]
```diff
@@ -163,7 +166,42 @@ class ConfidenceInterval:
...
+            The result of running minimize().
+        p_names : list, optional
+            Names of the parameters for which the CI is calculated. If None
+            (default), the CI is calculated for every parameter.
+        prob_func : None or callable, optional
+            Function to calculate the probability from the optimized chi-square.
+            Default is None and uses the built-in function `f_compare`
+            (i.e., F-test).
+        sigmas : list, optional
+            The sigma-levels to find (default is [1, 2, 3]). See Notes below.
```
Comment (reneeotten) [inline lmfit/confidence.py:187]: remove the "See Notes below."

File: lmfit/confidence.py [hunk 3]
```diff
@@ -163,7 +166,42 @@ class ConfidenceInterval:
...
+        trace : bool, optional
+            Defaults to False; if True, each result of a probability
+            calculation is saved along with the parameter. This can be used to
+            plot so-called "profile traces".
+        verbose : bool, optional
+            Print extra debugging information (default is False).
+        maxiter : int, optional
+            Maximum of iteration to find an upper limit (default is 200).
```
Comment (reneeotten) [inline lmfit/confidence.py:195]: it appears that the default is "50" instead of "200"?

File: lmfit/confidence.py [hunk 4]
```diff
@@ -163,7 +166,42 @@ class ConfidenceInterval:
...
+        min_rel_change : float, optional
+            Minimum relative change in probability (default is 1e-5).
+
+        Raises
+        ------
+        MinimizerException
+            If there are less than 2 variables or if the stderrs are not
```
Comment (reneeotten) [inline lmfit/confidence.py:202]: "2" --> "two" (it's a number below ten and should be written out)

File: lmfit/confidence.py [hunk 5]
```diff
@@ -163,7 +166,42 @@ class ConfidenceInterval:
...
+            sensible.
+        """
```
Comment (reneeotten) [inline lmfit/confidence.py:204]: should be a newline before closing the docstring

**Review by reneeotten — CHANGES_REQUESTED** [review]
thanks @Anselmoo ; it looks like that correcting the position of the new argument fixed the tests as well. This looks good to me, except for a few changes that are required in the docstring.

**Comment by newville** [comment]
@Anselmoo Thanks - that looks good to me.

**Pull request merged**